tests/test_cases/test_closedown/test_closedown.py

```diff
@@ -57,12 +51,20 @@ def clock_mon(dut):
 
 
 @cocotb.test(expect_fail=True,
-             expect_error=cocotb.SIM_NAME.lower().startswith(("modelsim",  # $fail_test() fails hard on Questa
+             expect_error=cocotb.SIM_NAME.lower().startswith(("modelsim",  # $fatal() fails hard on Questa
                                                               )))
 def test_failure_from_system_task(dut):
-    """Allow the dut to call $fail_test() from verilog"""
-    clock = Clock(dut.clk, 100)
-    clock.start()
+    """Allow the dut to call system tasks from verilog.
+    $fatal() will fail the test, and scheduler will cleanup forked coroutines.
+    """
+    cocotb.fork(Clock(dut.clk, 100).start())
     cocotb.fork(clock_mon(dut))
     cocotb.fork(run_external(dut))
-    yield Timer(10000000)
+    yield Timer(10000000, units='ns')
```

Comment:
Why don't you make this `async`/`await` now. It's on our Python 2 cleanup todo list.